File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution 
{
    public long solution(long n) 
    {
        string s = n.ToString();
        List<char> c = new List<char>(s.ToList());
        
        List<char> ans = c.OrderByDescending(x => x).ToList();
        
        string a = new string(ans.ToArray());
        long result = long.Parse(a);
        return result;
    }
}

File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

long long solution(long long n) 
{
    string strN = to_string(n);
    
    vector<int> vec;
    for(char &c : strN)
    {
        vec.push_back(c-'0');
    }
    sort(vec.begin(), vec.end(), greater<int>());
    
    string result = "";
    for(int &i : vec)
    {
        result += to_string(i);     
    }
    
    return stoll(result);
}